### PR TITLE
Use Popen in run_cmd and pipe outputs to logger

### LINKED ISSUE #211: run_via_binary,run_command: pipe outputs
we should pipe stdout and stderr of run_command and log it, instead of printing to output directly so colin has prettier output:

no
```
3e357036cbf2ac54d6bc62314e563608c125a8e443366f5f047dab88c7cb3606
PASS:Shell has to be runnable.

PASS:1 
```

yes
```
PASS:Shell has to be runnable.

PASS:1 
```

### PR DESCRIPTION
**Pull request by SkullTech**
Fixes #211 

_Old behaviour_

```
>>> from conu.utils import run_cmd
>>> ret = run_cmd(['ls'])
CHANGELOG.md	  docs			pytest.ini	   setup.cfg
CONTRIBUTING.md   examples		README.md	   setup.py
conu		  LICENSE		release-conf.yaml  test-requirements.sh
conu.spec	  Makefile		requirements.in    tests
Dockerfile	  MANIFEST.in		requirements.sh    Vagrantfile
Dockerfile.tests  pytest-container.ini	requirements.txt
>>> ret
>>> ret = run_cmd(['ls'], return_output=True)
>>> ret
'CHANGELOG.md\nCONTRIBUTING.md\nconu\nconu.spec\nDockerfile\nDockerfile.tests\ndocs\nexamples\nLICENSE\nMakefile\nMANIFEST.in\npytest-container.ini\npytest.ini\nREADME.md\nrelease-conf.yaml\nrequirements.in\nrequirements.sh\nrequirements.txt\nsetup.cfg\nsetup.py\ntest-requirements.sh\ntests\nVagrantfile\n'
```

_New behaviour_

```
>>> from conu.utils import run_cmd
>>> ret = run_cmd(['ls'])
>>> ret
>>> ret = run_cmd(['ls'], return_output=True)
>>> ret
'CHANGELOG.md\nCONTRIBUTING.md\nconu\nconu.spec\nDockerfile\nDockerfile.tests\ndocs\nexamples\nLICENSE\nMakefile\nMANIFEST.in\npytest-container.ini\npytest.ini\nREADME.md\nrelease-conf.yaml\nrequirements.in\nrequirements.sh\nrequirements.txt\nsetup.cfg\nsetup.py\ntest-requirements.sh\ntests\nVagrantfile\n'
```

### CODE REVIEW

**Comment by jpopelka** [comment]
[test]

File: conu/utils/__init__.py [hunk 1]
```diff
@@ -123,13 +123,13 @@ def run_cmd(cmd, return_output=False, ignore_status=False, **kwargs):
...
+                                   universal_newlines=True, **kwargs)
+        output = process.communicate()[0]
+        logger.debug(output)
         if return_output:
-            return subprocess.check_output(cmd,
-                                           stderr=subprocess.STDOUT,
-                                           universal_newlines=True,
-                                           **kwargs)
-        else:
-            subprocess.check_call(cmd, **kwargs)
```
Comment (TomasTomecek) [inline conu/utils/__init__.py]: you should also check for return code of the command; if it's `> 0`, an exception should be raised

File: conu/utils/__init__.py [hunk 2]
```diff
@@ -123,13 +123,13 @@ def run_cmd(cmd, return_output=False, ignore_status=False, **kwargs):
     """
     logger.debug('command: "%s"' % ' '.join(cmd))
     try:
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT,
+                                   universal_newlines=True, **kwargs)
+        output = process.communicate()[0]
```
Comment (TomasTomecek) [inline conu/utils/__init__.py]: I was hoping that communicate would iteratively print output after every newline; unfortunately, it returns all at once.

I guess it's fine for now, we can improve it later.

As a good example here is:
```
run_cmd(["bash", "-c", "for x in `seq 1 10`; do echo $x; sleep 1; done"])
```

File: conu/utils/__init__.py (hunk 1 above)
Comment (SkullTech) [reply conu/utils/__init__.py]: Okay got it. That makes sense.

File: conu/utils/__init__.py [hunk 3]
```diff
@@ -122,22 +122,21 @@ def run_cmd(cmd, return_output=False, ignore_status=False, **kwargs):
...
+    logger.debug(output)
+
+    if process.returncode > 0:
         if ignore_status:
             if return_output:
-                return cpe.output
+                return output
             else:
-                return cpe.returncode
+                return process.returncode
```
Comment (TomasTomecek) [inline conu/utils/__init__.py]: this is a bit confusing to me, but we can leave it for now

**Comment by TomasTomecek** [comment]
[test]

File: tests/integration/test_utils.py [hunk 4]
```diff
@@ -60,3 +60,20 @@ def test_command_exists():
 
 def test_check_docker():
     assert check_docker_command_works()
+
+
+def test_run_cmd():
+    ret = run_cmd(["sh", "-c", "for x in `seq 1 5`; do echo $x; sleep 1; done"])
```
[the remark this answers is not in the document]
Comment (SkullTech) [reply tests/integration/test_utils.py]: Ah yeah of course. On it.

**Comment by TomasTomecek** [comment]
moar [test]

**Comment by TomasTomecek** [comment]
Once the CI is green, we can merge; nicely done!

**Pull request merged**